html2/tokenizer.cpp

```diff
@@ -214,6 +236,507 @@ void Tokenizer::run() {
                 }
             }
 
+            case State::ScriptDataLessThanSign: {
+                auto c = consume_next_input_character();
+                if (!c) {
+                    emit(CharacterToken{'<'});
+                    reconsume_in(State::ScriptData);
+                    continue;
+                }
+
+                switch (*c) {
+                    case '/':
+                        temporary_buffer_ = "";
+                        state_ = State::ScriptDataEndTagOpen;
+                        continue;
+                    case '!':
+                        state_ = State::ScriptDataEscapeStart;
+                        emit(CharacterToken{'<'});
+                        emit(CharacterToken{'!'});
+                        continue;
+                    default:
+                        emit(CharacterToken{'<'});
+                        reconsume_in(State::ScriptData);
+                        continue;
+                }
+            }
+
+            case State::ScriptDataEndTagOpen: {
+                auto c = consume_next_input_character();
+                if (c && is_ascii_alpha(*c)) {
+                    current_token_ = EndTagToken{};
+                    reconsume_in(State::ScriptDataEndTagName);
+                    continue;
+                }
+
+                emit(CharacterToken{'<'});
+                emit(CharacterToken{'/'});
+                reconsume_in(State::ScriptData);
+                continue;
+            }
+
+            case State::ScriptDataEndTagName: {
+                auto c = consume_next_input_character();
+
+                auto anything_else = [this] {
+                    emit(CharacterToken{'<'});
+                    emit(CharacterToken{'/'});
+                    emit_temporary_buffer_as_character_tokens();
+                    reconsume_in(State::ScriptData);
+                };
+
+                if (!c) {
+                    anything_else();
+                    continue;
+                }
+
+                if (is_ascii_upper_alpha(*c)) {
+                    auto lower_case_c = to_lower(*c);
+                    std::get<EndTagToken>(current_token_).tag_name.append(1, lower_case_c);
+                    temporary_buffer_.append(1, lower_case_c);
```

Comment:
"Append the current input character to the temporary buffer.", so this probably shouldn't be lower_case_c?
https://html.spec.whatwg.org/multipage/parsing.html#script-data-end-tag-name-state

Reply:
Good catch! Fixed!